api/v1/ytsaurus_webhook.go

```diff
@@ -398,6 +398,31 @@ func (r *Ytsaurus) validateYQLAgents(*Ytsaurus) field.ErrorList {
 	return allErrors
 }
 
+func (r *Ytsaurus) validateUi(*Ytsaurus) field.ErrorList {
+	var allErrors field.ErrorList
+
+	if r.Spec.UI != nil {
+		if r.Spec.UI.UseInsecureCookies != nil && !*r.Spec.UI.UseInsecureCookies && !r.Spec.UI.Secure {
+			allErrors = append(allErrors, field.Invalid(field.NewPath("spec", "ui", "useInsecureCookies"), r.Spec.UI.UseInsecureCookies, "useInsecureCookies is deprecated, use secure instead"))
+		}
+		if r.Spec.UI.Secure {
+			for i, hp := range r.Spec.HTTPProxies {
+				if hp.Role != consts.DefaultHTTPProxyRole {
+					continue
+				}
+				if hp.Transport.HTTPSSecret == nil {
+					allErrors = append(allErrors, field.Required(
+						field.NewPath("spec", "httpProxies").Index(i).Child("transport", "httpsSecret"),
+						fmt.Sprintf("configured HTTPS for proxy with `%s` role is required for ui.secure", consts.DefaultHTTPProxyRole)))
+				}
+				break
+			}
+		}
+	}
+
+	return allErrors
+}
```

Comment:
On the second thought, I think we break some user experience here. 
Before the change user could choose if he wants https or http cookie for frontend (it being set via YT_AUTH_ALLOW_INSECURE) and after the change user can either use http-cookie or if he wants https-cookie he have to configure https for proxies.
And this is two different things which are now depend on one setting.

I suggest we un-deprecate `UseInsecureCookies` settings and let it configure cookies and `Secure` would configure https for UI <-> proxy communication.

@koct9i @sgburtsev am I missing something?

Reply:
Oh, I see. You are right. I'll revert it in a separate PR.